src/dev/flang/ast/ResolvedNormalType.java

```diff
@@ -646,6 +651,31 @@ AbstractFeature originalOuterFeature(AbstractFeature currentOuter)
       };
   }
 
+
+  /**
+   * traverse a resolved type collecting all features this type uses.
+   *
+   * @param s the features that have already been found
+   */
+  protected void usedFeatures(Set<AbstractFeature> s)
+  {
+    var f = featureOfType();
```

Comment:
This currently does not touch the outer features. This means that for a type like `(x T).y U` the visibility of `x` and `T` will be ignored, which is probably wrong. 

But this can be fixed later.